modules/utils.lua

```diff
@@ -142,6 +142,26 @@ function GenerateRandomPlate(pattern) -- luacheck: ignore
     return newPattern:upper()
 end
 
+--- maps a table by the given subfield
+--- @param subfield string
+--- @param table table
+--- @return table
+function MapTableBySubfield(subfield, table)
+	local mappedTable = {}
+	for tableIndex, tableData in pairs(table) do
+		if not tableData[subfield] then
+			goto continue
+		end
+		if not mappedTable[tableData[subfield]] then
+			mappedTable[tableData[subfield]] = {}
+		end
+
+		mappedTable[tableData[subfield]][tableIndex] = tableData
```

Comment:
What happens if table is a map rather than an array?

Reply:
I guess it would return itself?

Reply:
If what you mean is can i map a table, and then re-map that, the answer is no.
since we're mapping arrays, there wouldn't be any subfield to select.